lib/service_check.pm

```diff
@@ -116,14 +124,31 @@ our $default_services = {
     },
 };
 
+#
+# use subroutine to call the function with parameters
+#
+sub call_func {
+    # assuming that $f is a function reference
+    my $f = shift;
+    $f->(@_);
+}
+
 sub install_services {
     my $service = shift;
```

Comment:
I would change this to `my ($service) = @_;` as you are doing below in apache_service_check().